### ISSUE TITLE
Ohai plugin_path comparison does not handle different path separators 

### ISSUE THREAD

**Issue by Tech356** [chef-cookbooks/ohai#58]
### Cookbook version
4.2.3

### Chef-client version
12.19.36

### Platform Details
Windows 7 x64

### Scenario:
Set `ohai.plugin_path` to `c:\chef\ohai\plugins` using Windows style path separators. 

### Steps to Reproduce:
Add `ohai.plugin_path << 'c:\chef\ohai\plugins'` to the client.rb configuration file. 

### Expected Result:
I expect the configuration value work and not cause warnings. 

### Actual Result:
Ohai reads the configuration settings and runs the custom plugin(s). When Chef gets to the `ohai_plugin` resource, it thinks it needs to reload the plugin. That process throws multiple warnings. 
```
INFO: Processing ohai_plugin[my_plugin_1] action create (my_cookbook::default line 12)
WARN: The Ohai plugin_path does not include C:/chef/ohai/plugins. Ohai will reload on each chef-client run in order to add this directory to the path unless you modify your client.rb configuration to add this directory to plugin_path. The plugin_path can be set via the chef-client::config recipe. See 'Ohai Settings' at https://docs.chef.io/config_rb_client.html#ohai-settings for more details.
WARN: Adding C:/chef/ohai/plugins to the Ohai plugin path for this chef-client run only
INFO: Processing directory[C:/chef/ohai/plugins] action create (C:/chef/cache/cookbooks/ohai/resources/plugin.rb line 80)
INFO: Processing cookbook_file[C:/chef/ohai/plugins/my_plugin_1.rb] action create (C:/chef/cache/cookbooks/ohai/resources/plugin.rb line 87)
INFO: Processing ohai[my_plugin_1] action reload (C:/chef/cache/cookbooks/ohai/resources/plugin.rb line 110)
WARN: Plugin Definition Error: <C:/Chef/ohai/plugins/my_plugin_1.rb>: collect_data already defined on platform windows
WARN: Plugin Definition Error: <C:/Chef/ohai/plugins/my_plugin_2.rb>: collect_data already defined on platform windows
WARN: Plugin Definition Error: <C:/Chef/ohai/plugins/my_plugin_3.rb>: collect_data already defined on platform windows
WARN: Plugin Definition Error: <C:/Chef/ohai/plugins/my_plugin_4.rb>: collect_data already defined on platform windows
WARN: Plugin Definition Error: <C:/Chef/ohai/plugins/my_plugin_5.rb>: collect_data already defined on platform windows
WARN: Plugin Definition Error: <C:/Chef/ohai/plugins/my_plugin_6.rb>: collect_data already defined on platform windows
WARN: Plugin Definition Error: <C:/Chef/ohai/plugins/my_plugin_7.rb>: collect_data already defined on platform windows
INFO: ohai[my_plugin_1] reloaded
```

From what I can tell, when the Ohai cookbook checks to see if a plugin is already in the path, it simply compares strings and does not take into account different types of path separators. 

https://github.com/chef-cookbooks/ohai/blob/58b596c7c0d0e6dd15a59d37567bac02bb12c26c/resources/plugin.rb#L50

The simple workaround is to use `/` as the path separator in client.rb file. The ohai settings [documentation](https://docs.chef.io/config_rb_client.html#ohai-settings) does not specific why types of path separators are allowed.

**Comment by tas50** [comment]
We could certainly make this more robust for Windows users.  Thanks for the report

**Comment by tas50** [comment]
I've created a branch that reworks how the helper methods work and adds additional testing. It includes a new helper that normalizes the slashes so we can properly do a comparison. It turns out comparing Windows paths is actually a bit tricky if the directory doesn't necessarily exist on the filesystem yet, which is true in our case.

https://github.com/chef-cookbooks/ohai/pull/70

**Comment by tas50** [comment]
This *should* be fixed in 5.2.0. I've added a ton of testing so we can make sure this sort of thing doesn't crop up in the future as well
